### PR TITLE
SWATCH-2790: Use saveOrUpdate when syncing subscriptions

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: SWATCH-2790

## Description
The method `persist` will always perform an INSERT in database and this is causing issues when there are subscriptions with no subscription number, but the subscription ID correctly sets (it sounds weird, but it's what is happening in stage).

Using the method `saveOrUpdate` will correctly update the existing subscription if exists (so the subscription without the subscription number will be correctly fixed).

Note that because the `persist` method is a bit problematic (it happened the same when synchorinizing offerings), I replaced all the methods to `saveOrUpdate`. 

Apart from these changes:
- I created the stub search API which is by default false, because it was very convenient to reproduce the issue locally
- And disable the channels that consumed messages from umb if the property UMB_ENABLED is false: this is to avoid having a lot of exceptions locally

## Testing
1.- podman_compose up
2.- ./gradlew liquibaseUpdate
3.- load the stage data into your local database
4.- check that the following subscription is missing the subscription number:

```
select * from subscriptions where subscription_id ='6438190'
```

The subscription_number is empty.

5.- start the contracts service using the stubs: `SUBSCRIPTION_CLIENT_USE_STUB=true ./gradlew :swatch-contracts:quarkusDev`
6.- try to sync the problematic subscription:

```
<CanonicalMessage xmlns="http://esb.redhat.com/Canonical/6">
    <Payload>
        <Sync>
            <Subscription>
                <Identifiers>
                    <MasterSystem>SUBSCRIPTION</MasterSystem>
                    <Identifier entity-name="Install Base"
                        qualifier="number" system="EBS">39832705</Identifier>
                    <Identifier entity-name="Registration" qualifier="number">e401bce70edd43bd</Identifier>
                    <Identifier entity-name="Subscription"
                        qualifier="number" system="SUBSCRIPTION">6438195</Identifier>
                    <Reference entity-name="Installation" qualifier="number">c967ef3956ff3def</Reference>
                    <Reference entity-name="Customer" qualifier="number" system="EBS">customer123</Reference>
                    <Reference entity-name="Account" qualifier="number" system="EBS">account123</Reference>
                    <Reference entity-name="Customer" qualifier="id" system="WEB">7746627</Reference>
                    <Reference entity-name="Account" qualifier="id" system="WEB">16764251</Reference>
                </Identifiers>
                <Status>
                    <State>Active</State>
                    <StartDate>2023-04-24T00:00:00.000</StartDate>
                </Status>
                <Quantity>1</Quantity>
                <effectiveStartDate>2019-08-05T00:00:00.000</effectiveStartDate>
                <effectiveEndDate>2024-08-04T00:00:00.000</effectiveEndDate>
                <Product serviceable="false">
                    <CreatedBy>OSUBPRODTEST3EAC72C765804E2B8D34484E7FF8D1C8</CreatedBy>
                    <CreatedDate>2023-04-24T14:11:16.809</CreatedDate>
                    <Identifiers>
                        <MasterSystem>EBS</MasterSystem>
                        <Identifier entity-name="Product"
                            qualifier="number" system="SUBSCRIPTION">6438195</Identifier>
                    </Identifiers>
                    <Sku>RH00129F5</Sku>
                    <InventoryOperatingUnit>
                        <Number>118</Number>
                    </InventoryOperatingUnit>
                    <Product serviceable="true">
                        <CreatedBy>OSUBPRODTEST3EAC72C765804E2B8D34484E7FF8D1C8</CreatedBy>
                        <CreatedDate>2023-04-24T14:11:16.809</CreatedDate>
                        <LastUpdatedBy>SYSADMIN</LastUpdatedBy>
                        <LastUpdatedDate>2023-04-24T14:17:02.000</LastUpdatedDate>
                        <Identifiers>
                            <MasterSystem>EBS</MasterSystem>
                            <AuthoringOperatingUnit>
                                <Number>103</Number>
                            </AuthoringOperatingUnit>
                            <Identifier entity-name="Product"
                                qualifier="number" system="SUBSCRIPTION">37188550</Identifier>
                            <Reference entity-name="Contract"
                                qualifier="number" system="EBS">16612423</Reference>
                            <Reference entity-name="Contract"
                                qualifier="id" system="EBS">25565753</Reference>
                        </Identifiers>
                        <Status>
                            <State>Active</State>
                            <StartDate>2023-04-24T00:00:00.000</StartDate>
                            <EndDate>2026-04-23T00:00:00.000</EndDate>
                        </Status>
                        <Status>
                            <State>Signed</State>
                            <StartDate>2023-04-24T00:00:00.000</StartDate>
                        </Status>
                        <Status primary="true">
                            <State>Terminated</State>
                            <StartDate>2023-04-24T14:17:02.000</StartDate>
                        </Status>
                        <Sku>SVC2681</Sku>
                        <InventoryOperatingUnit>
                            <Number>118</Number>
                        </InventoryOperatingUnit>
                        <ContractDescription>Reg Number activation from Hock</ContractDescription>
                        <ContractHeaderStatus>Terminated</ContractHeaderStatus>
                    </Product>
                </Product>
            </Subscription>
        </Sync>
    </Payload>
</CanonicalMessage>
```

7.- verify that the call worked ok and the subscription entity was updated:

```
select * from subscriptions where subscription_id ='6438190'
```

The subscription_number should be `6438195`.

### CODE REVIEW

File: swatch-contracts/src/main/java/com/redhat/swatch/contract/service/SubscriptionSyncService.java [hunk 1]
```diff
@@ -75,7 +75,6 @@ public class SubscriptionSyncService {
   private final ProductDenylist productDenylist;
   private final EntityManager entityManager;
 
-  @Transactional
```
Comment (Sgitario) [inline swatch-contracts/src/main/java/com/redhat/swatch/contract/service/SubscriptionSyncService.java:78]: Same as in https://github.com/RedHatInsights/rhsm-subscriptions/pull/3592/files#r1704022345.

**Comment by kartikshahc** [comment]
@Sgitario don't you have to also change contract flow > subscription update? Won't the persist have same issue here?
https://github.com/RedHatInsights/rhsm-subscriptions/blob/996fbd2c92f4c24a7f439a165d2eabf76e876ca3/swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java#L416

**Review by kartikshahc — CHANGES_REQUESTED** [review]
It looks good to me except the comment I have for contract>subscription sync

**Comment by Sgitario** [comment]
> @Sgitario don't you have to also change contract flow > subscription update? Won't the persist have same issue here?
> 
> https://github.com/RedHatInsights/rhsm-subscriptions/blob/996fbd2c92f4c24a7f439a165d2eabf76e876ca3/swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java#L416

Very well spotted! Will change it.

**Comment by Sgitario** [comment]
@kartikshahc I've already addressed your comments, and I'm merging this, so stage is fixed.

**Pull request merged**